packages/edit-site/src/components/dataviews/add-filter.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	privateApis as componentsPrivateApis,
+	Button,
+	BaseControl,
+	Icon,
+} from '@wordpress/components';
+import { chevronRightSmall, plus } from '@wordpress/icons';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../../lock-unlock';
+
+const {
+	DropdownMenuV2,
+	DropdownSubMenuV2,
+	DropdownSubMenuTriggerV2,
+	DropdownMenuItemV2,
+} = unlock( componentsPrivateApis );
+
+export default function AddFilter( { filters, onChangeView } ) {
+	if ( ! filters.length ) {
+		return (
+			<BaseControl>
+				<Button
+					prefix={ <Icon icon={ plus } /> }
+					disabled={ ! filters?.length }
+					variant="tertiary"
+				>
+					{ __( 'Add filter' ) }
+				</Button>
+			</BaseControl>
+		);
+	}
+
+	return (
+		<BaseControl>
+			<DropdownMenuV2
+				label={ __( 'Add filter' ) }
+				trigger={
+					<Button variant="tertiary">{ __( '+ Add filter' ) }</Button>
+				}
+			>
+				{ filters.map( ( filter ) => (
+					<DropdownSubMenuV2
+						key={ filter.field }
+						trigger={
+							<DropdownSubMenuTriggerV2
+								suffix={ <Icon icon={ chevronRightSmall } /> }
+							>
+								{ filter.name }
+							</DropdownSubMenuTriggerV2>
+						}
+					>
+						{ filter.elements.map( ( element ) => (
+							<DropdownMenuItemV2
+								key={ element.value }
+								onSelect={ () => {
+									onChangeView( ( currentView ) => ( {
+										...currentView,
+										filters: [
+											...currentView.filters,
+											{
+												field: filter.field,
+												operator: 'in',
```

Comment:
Do all fields support "in" operators? Maybe for some filters, you can't "select" a value to add them but instead add an "input" as soon as you click on them (no nested dropdown for them). 

I see different fields having different behaviors in this dropdown but maybe it's for later.